src/rpft/parsers/creation/contentindexparser.py

```diff
@@ -24,29 +24,29 @@ def __init__(self, table, argument_definitions):
 class ContentIndexParser:
     def __init__(
         self,
-        sheet_reader=None,
+        sheet_reader,
         user_data_model_module_name=None,
         tag_matcher=TagMatcher(),
     ):
+        self.sheet_reader = sheet_reader
         self.tag_matcher = tag_matcher
         self.template_sheets = {}  # values: tablib tables
         self.data_sheets = {}  # values: OrderedDicts of RowModels
         self.flow_definition_rows = []  # list of ContentIndexRowModel
-        self.campaign_parsers = []  # list of CampaignParser
+        self.campaign_parsers = {}  # list of CampaignParser
         if user_data_model_module_name:
             self.user_models_module = importlib.import_module(
                 user_data_model_module_name
             )
-        if sheet_reader:
-            self.add_content_index(sheet_reader)
+        main_sheets = self.sheet_reader.get_main_sheets()
+        if not main_sheets:
+            LOGGER.critical("No content index sheet provided")
+        # TODO: add name of file that the sheet is from?
```

Comment:
No todos, please. Create an issue if you think it's a worthwhile task.

Reply:
Implemented now because the refactor made it easy.